src/dict.c

```diff
@@ -123,133 +130,163 @@ int _dictInit(dict *d, dictType *type)
 
 /* Resize the table to the minimal size that contains all the elements,
  * but with the invariant of a USED/BUCKETS ratio near to <= 1 */
-int dictResize(dict *d)
-{
+int dictResize(dict *d) {
     unsigned long minimal;
 
     if (!dict_can_resize || dictIsRehashing(d)) return DICT_ERR;
-    minimal = d->ht_used[0];
+    minimal = d->ht_used;
     if (minimal < DICT_HT_INITIAL_SIZE)
         minimal = DICT_HT_INITIAL_SIZE;
-    return dictExpand(d, minimal);
+    return _dictResize(d, minimal, NULL);
 }
 
-/* Expand or create the hash table,
- * when malloc_failed is non-NULL, it'll avoid panic if malloc fails (in which case it'll be set to 1).
- * Returns DICT_OK if expand was performed, and DICT_ERR if skipped. */
-int _dictExpand(dict *d, unsigned long size, int* malloc_failed)
-{
+/* Resize or create the hash table. When malloc_failed is non-NULL, it'll avoid
+ * panic if malloc fails (in which case it'll be set to 1).
+ * Returns DICT_OK if resize was performed, and DICT_ERR if skipped. */
+int _dictResize(dict *d, unsigned long size, int *malloc_failed) {
     if (malloc_failed) *malloc_failed = 0;
 
     /* the size is invalid if it is smaller than the number of
      * elements already inside the hash table */
-    if (dictIsRehashing(d) || d->ht_used[0] > size)
+    if (dictIsRehashing(d) || d->ht_used > size)
         return DICT_ERR;
 
-    /* the new hash table */
+    /* the new reallocated hash table */
     dictEntry **new_ht_table;
-    unsigned long new_ht_used;
     signed char new_ht_size_exp = _dictNextExp(size);
 
     /* Detect overflows */
-    size_t newsize = 1ul<<new_ht_size_exp;
-    if (newsize < size || newsize * sizeof(dictEntry*) < newsize)
+    size_t newsize = 1ul << new_ht_size_exp;
+    if (newsize < size || newsize * sizeof(dictEntry *) < newsize)
         return DICT_ERR;
 
     /* Rehashing to the same table size is not useful. */
-    if (new_ht_size_exp == d->ht_size_exp[0]) return DICT_ERR;
-
-    /* Allocate the new hash table and initialize all pointers to NULL */
-    if (malloc_failed) {
-        new_ht_table = ztrycalloc(newsize*sizeof(dictEntry*));
-        *malloc_failed = new_ht_table == NULL;
-        if (*malloc_failed)
-            return DICT_ERR;
-    } else
-        new_ht_table = zcalloc(newsize*sizeof(dictEntry*));
-
-    new_ht_used = 0;
+    if (new_ht_size_exp == d->ht_size_exp) return DICT_ERR;
+
+    /* If hash table grows - then realloc() bigger memory. */
+    /* If first init - then this condition still TRUE. realloc() memory */
+    if (d->ht_size_exp < new_ht_size_exp) {
+        /* Realloc the hash table (without initializing new buckets to NULL) */
+        if (malloc_failed) {
+            new_ht_table = ztryrealloc(d->ht_table,
+                                       newsize * sizeof(dictEntry *));
```

Comment:
It seems that it doesn't need 2 lines(only 84 columns).

Reply:
Generally speaking, it is too bad that Redis don't have code convention - maybe about time to have such one. We better have a Redis convention that will be applied, at least for new files, or files with major change.

I usually try to follow the convention that applied to the file I am trying to modify and in this file it is up to 80.

Reply:
I have to agree :)

IMO, we better not pass behind 100 lines because of the "half screen rule" (invented it just right now. lol)
I will change it. thank you.

Reply:
@moticless FYI: I support the half-screen rule! With side-by-side Emacs and terminal on my laptop, with a sufficiently large font size, that means 80 columns fit perfectly well, while 100+ columns means I have to scroll sideways.

Reply:
i do much of my work on [this](https://store.planetcom.co.uk/products/cosmo-communicator) device, and i'd like to fit 3 (not two), files side by side..
that said, i carry a large magnifier glass in my pocket at any time, so i don't mind tiny fonts :stuck_out_tongue:

Reply:
@zuidetkwast I prefer to be able to see more columns in one screen because my terminal is below. 8-(

Reply:
@sundb i suppose you mean rows (lines, not columns).
this is exactly my point.. wasting too many lines on nonsense (and error handling), make it hard to see the big picture.
our editors have too many toolbars and panes on the top and bottom, and not many lines fit in the screen.
on the other hand, the horizontal space is abundant and mostly unused.
if you wanna see things side by side, get another monitor :stuck_out_tongue: 

my other point is that any hard number would be wrong.
i'd argue that 100 is too much for my environment, and i'd like to to be 96 (or 128).

my point is each piece of code should be judged separately and written in a way that it's most readable.
breaking line at an **arbitrary** position often make the code very hard to read.

Reply:
My rule of thumb has been:
1. If it's over 80, it's probably too long and try to simplify as long as it doesn't impact readability. Look for stuff like simpler names or reducing indentation in functions. Never sacrifice readability here.
2. If it's over 100, it's okay to sacrifice a bit of readability to bring stuff down.
3. If it's over 120, it's too long, you should break it up. It's always a pain to review.

Reply:
@sundb Do you have a monitor which can be rotated? Have you tried rotating it 90 degrees and use portrait mode? (I know some people who do this.)

Reply:
@zuiderkwast I can but I don't like to do it, because with only a 1080p monitor, it's too crowded vertically (also have a code tree on the left).
My rule is that if the code is exactly over 80, but not too much, I don't break the line.